10 link story api

charlottebrf opened the pull request:
Closes #1
Closes #10

RSijelmass committed:
Copies api spike from different branch, includes function to retreive headlines

RSijelmass committed:
Gets Guardian API in interface
- create getStories(), takes out api as string through xmlHttpRequest

charlottebrf committed:
Adds function createStories in Interface:
- gets Stories from API as a JSON
- rewritten Story to take in hash
- Story Component now not working in Interface

charlottebrf committed:
Headlines are now displayed:
-Fixes storySpec
-Creates loop in interface to createStory
-Updats headline in featureSpec

bannastre commented on spec/storySpec.js:
```diff
@@ -1,19 +1,14 @@
 }
 
-function hasText() {
-	var text = "This is a body!";
```
Why remove this test? We are still going to need this once we have the summary component?

bannastre commented on spec/storySpec.js:
```diff
@@ -1,19 +1,14 @@
   assert.isEqual(story.headline, headline);
 }
 
-function hasText() {
```
Why remove this test? We are still going to need this once we have the summary component?

bannastre commented on views/interface.js:
```diff
@@ -1,10 +1,26 @@
- renderStory = function(storyComponent) {
- 		storyComponent.render();
- }
+	function getApi(api) {
```
Can we do some better naming here? We have two api's to query so it might be good to have a distinction

bannastre commented on views/interface.js:
```diff
@@ -1,10 +1,26 @@
 
- renderStory(storyComponent);
+	var apiString = getApi(api);
+	var apiJson = JSON.parse(apiString);
```
Do we need JSON? We found that the results hash is already an object so doesn't need to be parsed.. may be different for Guardian api

bannastre commented on src/apiSpiking.js:
```diff
@@ -0,0 +1,9 @@
+var guardian = {"response":{"status":"ok","userTier":"developer","total":1960175,"startIndex":1,"pageSize":10,"currentPage":1,"pages":196018,"orderBy":"newest","results":[{"id":"sport/live/2017/jun/22/royal-ascot-2017-buildup-to-the-gold-cup-live","type":"liveblog","sectionId":"sport","sectionName":"Sport","webPublicationDate":"2017-06-22T14:28:41Z","webTitle":"Royal Ascot 2017: Benbatl wins Hamp ...
```
Do we need to have this file in the repo?

bannastre reviewed (commented):
Awesome work! I had a few questions so would be good to get together to understand your thinking

RSijelmass committed:
Rename argument in Story model
- rename function getApi to getGuardianApi
- also renames a few api variables in interface
- removes apiSpiking file

tobywinter reviewed (approved):
Good stuff, nailed the changes 
:shipit:
